src/App.jsx

```diff
@@ -21,11 +25,138 @@ function App() {
   }, []);
 
   return (
-    <>
-      <h1>{message}</h1>
-    </>
+    <BrowserRouter>
+      <nav>
+      <ul>
+        <li><Link to="/">Home</Link></li>
+        <li><Link to="/projects">Browse Projects</Link></li>
+        {!isLoggedIn && 
```

Comment:
We can simplify this conditional by coding it this way:

          {!isLoggedIn ? (
            <>
              <li>
                <Link to="/register">Join Now</Link>
              </li>
              <li>
                <Link to="/login">Sign In</Link>
              </li>
            </>
          ) : (
            <>
              <li>
                <Link to="/messaging">Messaging</Link>
              </li>
              <li>
                <Link to="/notification">Notification</Link>
              </li>
              <li>
                <Link to="/profile">Profile</Link>
              </li>
              <li>
                <Link to="/">Sign Out</Link>
              </li>
            </>
          )}